presto-main/src/test/java/com/facebook/presto/operator/scalar/TestJsonFunctions.java

```diff
@@ -317,4 +317,13 @@ public void testJsonEquality()
         assertFunction("JSON '{\"p_1\": 1, \"p_2\":\"v_2\", \"p_3\":null, \"p_4\":true, \"p_5\": {\"p_1\":1}}' != " +
                 "JSON '{\"p_2\":\"v_2\", \"p_4\":true, \"p_1\": 1, \"p_3\":null, \"p_5\": {\"p_1\":1}}'", BOOLEAN, false);
     }
+    
+    @Test
+    public void testScalarToJson()
+    {
+        assertFunction("JSON '1.234' = CAST(1.234 AS JSON)", BOOLEAN, true);
+        assertFunction("JSON '1.234' = CAST(CAST(1.234 AS REAL) AS JSON)", BOOLEAN, true);
+        assertFunction("JSON '10' = CAST(CAST(10 AS TINYINT) AS JSON)", BOOLEAN, true);
+        assertFunction("JSON '10000' = CAST(CAST(10000 AS SMALLINT) AS JSON)", BOOLEAN, true);
```

Comment:
Put these tests in `testCastFromIntegrals` or a new method that is called `testCastFromReal`